rust/connlib/snownet/src/node.rs

```diff
@@ -382,16 +357,39 @@ where
 
     /// Returns a pending [`Event`] from the pool.
     pub fn poll_event(&mut self) -> Option<Event<TId>> {
+        let binding_events = self.bindings.iter_mut().flat_map(|(server, binding)| {
+            iter::from_fn(|| binding.poll_event().map(|e| (*server, e)))
+        });
+        let allocation_events = self
+            .allocations
+            .iter_mut()
+            .flat_map(|(server, allocation)| {
+                iter::from_fn(|| allocation.poll_event().map(|e| (*server, e)))
+            });
+
+        for (server, event) in binding_events.chain(allocation_events) {
+            match event {
+                CandidateEvent::New(candidate) => {
+                    add_candidates(
+                        server,
+                        candidate,
+                        &mut self.connections,
+                        &mut self.pending_events,
+                    );
+                }
+            }
```

Comment:
```suggestion
            let CandidateEvent::New(candidate) = event;
            add_candidates(
                server,
                candidate,
                &mut self.connections,
                &mut self.pending_events,
            );
```

Reply:
Does that compile? I thought unreachable patterns are not yet stable. Plus this will need to be a `match` soon.